lib/src/debian.dart

```diff
@@ -0,0 +1,216 @@
+    ]);
+  });
+}
+
+/// Generate the control file for the Debian package.
+void _generateControlFile(String debianDir) {
+  var controlFilePath = p.join(debianDir, "DEBIAN", "control");
+  writeString(controlFilePath, controlData.value);
+}
+
+/// Scan for new .deb packages in the [repo] and update the `Packages` file.
+Future<void> _updatePackagesFile(String repo) async {
+  // Scan for new packages
+  String output = run("dpkg-scanpackages",
+      arguments: ["--multiversion", "."], workingDirectory: repo);
+  // Write the stdout to the file
+  writeString(p.join(repo, 'Packages'), output);
+  // Force Compress the Packages file
+  run("gzip", arguments: ["-k", "-f", "Packages"], workingDirectory: repo);
+}
+
+/// Generate the Release index for the PPA.
+Future<void> _updateReleaseFile(String repo) async {
+  String output = run("apt-ftparchive",
+      arguments: ["release", "."], workingDirectory: repo);
+  writeString(p.join(repo, 'Release'), output);
+}
+
+/// Sign the Release file with the GPG key.
+Future<void> _updateReleaseGPGFile(String repo) async {
+  String output = run("gpg",
+      arguments: [
+        "--default-key",
+        signingEmail.value,
+        "-abs",
+        "-o",
+        "-",
+        "Release",
+      ],
+      workingDirectory: repo);
+  writeString(p.join(repo, 'Release.gpg'), output);
+}
+
+/// Update the InRelease file with the new index and keys.
+Future<void> _updateInReleaseFile(String repo) async {
+  String output = run("gpg",
+      arguments: [
+        "--default-key",
+        signingEmail.value,
+        "--clearsign",
+        "-o",
+        "-",
+        "Release",
+      ],
+      workingDirectory: repo);
+  writeString(p.join(repo, 'InRelease'), output);
+}
+
+/// Commit all the changes in the PPA repository and push to the remote upstream.
+Future<void> _gitUpdateAndPush(String repo) async {
+  run("git",
+      arguments: ["add", "."],
+      workingDirectory: repo,
+      runOptions: botEnvironment);
+
+  run("git",
+      arguments: [
+        "commit",
+        "--all",
+        "--message",
+        "Update $humanName to $version"
+      ],
+      workingDirectory: repo,
+      runOptions: botEnvironment);
+
+  await runAsync("git",
+      arguments: [
+        "push",
+        url("https://$githubUser:$githubPassword@github.com/$debianRepo.git")
+            .toString(),
+        "HEAD:${await _originHead(repo)}"
+      ],
+      workingDirectory: repo);
+}
+
+/// Returns the name of HEAD i.e. default branch in the origin remote
+Future<String> _originHead(String repo) async {
+  var result = run("git",
+      arguments: ["symbolic-ref", "refs/remotes/origin/HEAD"],
+      workingDirectory: repo);
+
+  var prefix = "refs/remotes/origin/";
+  if (!result.startsWith(prefix)) {
+    fail('Unexpected output from "git symbolic-ref refs/remotes/origin/HEAD":\n'
+        'Expected a string starting with "$prefix", got:\n'
+        '$result');
+  }
+  return result;
+}
```

Comment:
This function does the same as the function `_originHead()` in `homebrew.md`, but that is a private function (prefixed with `_`). Shall I reuse that function directly in this module?

Reply:
Feel free to move any shared functions into `utils.dart`.

Reply:
Sure, thanks.